### PR TITLE
feat(completion): add complete_word and complete_type in v:event

### PR DESCRIPTION
**Pull request by glepnir**
adds key complete_word and complete_tye into v:event when CompleteDone event trigger

### CODE REVIEW

File: src/insexpand.c [hunk 1]
```diff
@@ -2296,13 +2296,72 @@ set_ctrl_x_mode(int c)
...
+	case CTRL_X_LOCAL_MSG:
+	    return (char_u *)"local-msg";        // Local message (internal)
+	case CTRL_X_EVAL:
+	    return (char_u *)"eval";             // Built-in function complete()
+	case CTRL_X_CMDLINE_CTRL_X:
+	    return (char_u *)"cmdline-ctrl-x";   // CTRL-X in command-line mode
+	default:
+	    return (char_u *)"unknown";          // Fallback for invalid mode
+    }
+}
```
Comment (zeertzjq) [inline src/insexpand.c]: This is inconsistent with the values in `:h complete_info_mode`

Comment (glepnir) [reply src/insexpand.c]: yep .. i think i can reuse them.

File: src/insexpand.c [hunk 2]
```diff
@@ -2296,13 +2300,75 @@ set_ctrl_x_mode(int c)
     return retval;
 }
 
+#if defined(FEAT_EVAL)
+    static char_u *
+ctrl_x_mode_to_string(int mode)
```
Comment (zeertzjq) [inline src/insexpand.c]: Why not reuse `ctrl_x_mode_names[]`?

Comment (glepnir) [reply src/insexpand.c]: Or modified ins_compl_mode ?

Comment (glepnir) [reply src/insexpand.c]: nvm that's not good

File: runtime/doc/autocmd.txt [hunk 3]
```diff
@@ -702,6 +702,12 @@ CompleteDone			After Insert mode completion is done.  Either
 				The |v:completed_item| variable contains
 				information about the completed item.
 
+				Sets these |v:event| keys:
+				    complete_word	The word that was
+							selected, empty if
+							abandoned complete.
+				    complete_type	See |ins-completion|
```
Comment (chrisbra) [inline runtime/doc/autocmd.txt]:
```suggestion
				    complete_type	See |complete_info_mode|
```

**Comment by chrisbra** [comment]
Thanks. I think looks good. 

What do you mean with:

> doc will update when we make sure add them in v:event

**Comment by glepnir** [comment]
nvm tried other way but not good.

**Comment by chrisbra** [comment]
thanks!

**Pull request closed without merging**